code/game/machinery/medical_kiosk.dm

```diff
@@ -0,0 +1,345 @@
+//The Medical Kiosk is designed to act as a low access alernative to  a medical analyzer, and doesn't require breaking into medical. Self Diagnose at your heart's content!
+//For a fee that is. Comes in 4 flavors of medical scan.
+
+
+/obj/machinery/medical_kiosk
+	name = "medical kiosk"
+	desc = "A freestanding medical kiosk, which can provide a wide range of medical analysis for diagnosis."
+	icon = 'icons/obj/machines/medical_kiosk.dmi'
+	icon_state = "kiosk"
+	layer = ABOVE_MOB_LAYER
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/medical_kiosk
+	payment_department = ACCOUNT_MED
+	var/obj/item/scanner_wand
+	var/default_price = 15          //I'm defaulting to a low price on this, but in the future I wouldn't have an issue making it more or less expensive.
+	var/active_price = 15           //Change by using a multitool on the board.
+	var/pandemonium = FALSE			//AKA: Emag mode.
+
+	var/scan_active_1 = FALSE       //Shows if the machine is being used for a general scan.
```

Comment:
You should refactor this into something like:
var/scan_state = SCAN_STATE_GENERAL
This is not fun to look at